pandas/tests/series/test_combine_concat.py

```diff
@@ -116,8 +116,32 @@ def test_update(self):
         df = DataFrame([{"a": 1}, {"a": 3, "b": 2}])
         df['c'] = np.nan
 
-        # this will fail as long as series is a sub-class of ndarray
-        # df['c'].update(Series(['foo'],index=[0])) #####
+        df['c'].update(Series(['foo'], index=[0]))
+        expected = DataFrame([[1, np.nan, 'foo'], [3, 2., np.nan]],
+                             columns=['a', 'b', 'c'])
+        assert_frame_equal(df, expected)
+
+    @pytest.mark.parametrize('other_values', [
+        [61, 63],       # int
+        [61., 63.],     # float, but can be cast to int
+        [61.1, 63.1],   # float, cannot be cast to int
+        [(61,), (63,)]  # object
+    ], ids=['int', 'float_castable', 'float', 'object'])
+    @pytest.mark.parametrize('caller_dtype', ['int', 'float', object])
+    def test_update_dtypes(self, caller_dtype, other_values):
+        caller_values = [10, 11, 12]
+        s = Series(caller_values, dtype=caller_dtype)
+        other = Series(other_values, index=[1, 3])
+        s.update(other)
+
+        expected_values = [caller_values[0], other_values[0], caller_values[2]]
+        try:
+            # we keep original dtype whenever possible
```

Comment:
pls don't use a try except inside the test itself. you should explicity use pytest.raises if you want to catch an exception

Reply:
@jreback 
I know, but this is fully intended (I've made the same argument to @gfyoung in a conversation he marked "resolved" below).

The thing is that `Series.update` will maintain the dtype of the caller as much as it can. This means, when calling from an `int` Series, floats get cast to `int`, *if possible*. Only if that doesn't work, is the dtype changed to whatever the updated values require.

If I were to pass expected results through the parametrization, I wouldn't be able to parametrize separately, and would therefore need 12 expected results. This is neither easy to read nor extensible.

Please have another look at this test, I think it's quite clear that the try-catch is not about skipping, but about different expected values (which I explained with the comment).